devito/operations/interpolators.py

```diff
@@ -185,30 +208,50 @@ def r(self):
         return self.sfunction.r
 
     @cached_property
-    def _rdim(self):
+    def _cdim(self):
+        """Base CustomDimensions used to construct _rdim"""
         parent = self.sfunction._sparse_dim
         dims = [CustomDimension("r%s%s" % (self.sfunction.name, d.name),
                                 -self.r+1, self.r, 2*self.r, parent)
                 for d in self._gdims]
+        return dims
+
+    @memoized_meth
+    def _rdim(self, subdomain=None):
+        # If the interpolation operation is limited to a SubDomain,
+        # use the SubDimensions of that SubDomain
+        if subdomain:
+            gdims = tuple(subdomain.dimension_map[d] for d in self._gdims)
+        else:
+            gdims = self._gdims
 
         # Make radius dimension conditional to avoid OOB
         rdims = []
         pos = self.sfunction._position_map.values()
 
-        for (d, rd, p) in zip(self._gdims, dims, pos):
+        for (d, rd, p) in zip(gdims, self._cdim, pos):
             # Add conditional to avoid OOB
             lb = sympy.And(rd + p >= d.symbolic_min - self.r, evaluate=False)
             ub = sympy.And(rd + p <= d.symbolic_max + self.r, evaluate=False)
             cond = sympy.And(lb, ub, evaluate=False)
+            # Insert a check to catch cases where interpolation/injection is
+            # into an empty rank. This depends on the injection field or interpolated
+            # expression, and so must be inserted here.
+            if subdomain and subdomain.distributor.is_parallel:
```

Comment:
this should only be needed for one condition not for all dims

Reply:
But you could make the condition only once and attach it to the sparse dimension instead of the radius to make it the outer condition

Reply:
But why bother?